Updated stop.ts

Levon-Volodin opened the pull request:
updated stop.ts so it checks for user rights instead of the SYSTEM flag which can possibly cause some unnecessary complications when it comes to instance management

*also has been tested with multiple tokens to be sure that this does NOT permit normal users to POST to the /stop API route
**NOTE**: instance owners will have to re-run `npm run setup` for these changes to take effect

Levon-Volodin committed:
Updated stop.ts

updated stop.ts so it checks for user rights instead of the SYSTEM flag which can possibly cause some unnecessary complications when it comes to instance management

*also has been tested with multiple tokens to be sure that this does NOT permit normal users to POST to the /stop API route
**NOTE**: instance owners will have to re-run `npm run setup` for these changes to take effect

erkinalp commented on api/src/routes/stop.ts:
```diff
@@ -5,21 +5,22 @@ import { User } from "@fosscord/util";
-	if(user.flags == '4096') {
+	//EXPERIMENTAL: have an "OPERATOR" platform permission implemented for this API route
+	const user = await User.findOneOrFail({ where: { id: req.user_id }, select: ["rights"] });
+	if(user.rights == '1') {
```
== 1 is wrong, you need to use bitwise operations; as rights can be combined

Levon-Volodin replied:
ah alright ill get back ASAP when these changes have been satisfied

Levon-Volodin committed:
Updated stop.ts

now using bitwise operators and dividing the `rights` value so you can have other rights whilst having the `OPERATOR` right

for example: `3` is `1` and `2` combined

NOTE: a potential issue has been uncovered while testing; if you have more than 1 thread the server will only stop 1 thread and the rest of the server will continue to run
